### ISSUE TITLE
AppleID keeps getting locked

### ISSUE THREAD

**Issue by schwa** [XcodesOrg/xcodes#365]
I keep getting the "your AppleID is locked" error when using Xcodes.

I know there have been a lot of problems with AppleID recently (the great account locking of late April…) but despite that this does seem to be more common in Xcodes and has certainly been happening before the issues AppleID had at the end of April.

For reference (and searchability) the error I get in Xcodes is: 

"Unable to install Xcode This Apple ID has been locked for security reasons. Visit Forgot to reset your account (https:///forgot.apple.com)."

And if I try to log in anyways I get a red warning under the login field ("This Apple ID has been locked for security reasons. Visit Forgot to reset your account (https:///forgot.apple.com).")

Is there anything Xcodes can do to mitigate these issues?

I am running Xcodes on 3 machines so I wonder (although I only launch the app when I know there is an Xcode to download) if that is causing the problem. I am unsure if Xcodes installs a persistent helper daemon (nothing in preferences implies it is).

Is it possible to switch from using my appleid and password to using "app specific" passwords?

Thanks

Screenshots attached.
<img width="285" alt="Screenshot 2024-05-08 at 09 21 48" src="https://github.com/XcodesOrg/xcodes/assets/20267/1ba7eeb7-35cd-4eb5-9b7a-02e911d22987">
<img width="441" alt="Screenshot 2024-05-08 at 09 22 53" src="https://github.com/XcodesOrg/xcodes/assets/20267/72e21844-31be-49b7-a665-c70618605c67">

**Comment by dfed** [comment]
Same here. Happened a few times in the last few weeks. Always when utilizing Xcodes.

**Comment by MattKiazyk** [comment]
> Is it possible to switch from using my appleid and password to using "app specific" passwords?

I would love love love if Xcodes could not be dependent on apple passwords and use app passwords, but every time I've tried, the 2 systems don't work with each other. ie The login token I get from X doesn't work with Y.  They way that Xcodes, Fastlane (and others) have always worked is by faking out the login you do at apple.com and fragile(ly) setting tokens and header keys. AppStoreConnect has a very nice api that supports this, but can't use that same api for developer.apple.com. 

Previously in the past, users were getting locked out, because Apple changed and added in a hashcash key https://github.com/XcodesOrg/XcodesApp/pull/361 . They may have well done something again, but I haven't seen anything widespread.

Hopefully all your machines @schwa @dfed  are on the latest version? Other than that, I'm not sure what else I can do to help this situation.

**Comment by dfed** [comment]
Yeah all my machines are on 2.1.2. The first time this happened there was [this more widespread incident](https://9to5mac.com/2024/04/26/signed-out-of-apple-id-account-problem-password/), and I've been getting my account lock relatively consistently when using Xcodes since. As a casual observer, it feels like Apple may have changed something here? That said, I'm far from in this world and I trust your judgement – just reporting my experience thus far.

**Comment by schwa** [comment]
Yeah latest version here.

Would extra logging help at least provide any insight into what's causing it?

Are you just displaying the error string some HTTP service gives you are you interpreting something else as a lock out? The lock out message is very specific - so I'm curious.

**Comment by schwa** [comment]
(Anecdotally other folks have mentioned this occurred to them this morning too - maybe somewhat widespread?)

**Comment by nalexn** [comment]
In my case it doesn't say apple id is locked, it simply rejects the correct 2FA code I enter. Tried like 5 times, same error: "The code that was entered is incorrect"

macOS 14.5 (23F79), M2; Xcodes 2.1.2(26)

<img width="436" alt="Screenshot 2024-05-14 at 15 52 57" src="https://github.com/XcodesOrg/xcodes/assets/1594746/740c0a35-dfae-4cb4-a522-77ef4ca6a506">

**Comment by schwa** [comment]
And this morning everything locked again - while rest of account on Mac is just fine.

Are there logs I can check to see what the responses are from Apple?

**Comment by dfed** [comment]
I haven't seen this in a bit – might have been a blip?

**Comment by vadimvitvickiy** [comment]
> I haven't seen this in a bit – might have been a blip?

For me, It's still an issue, cannot log in.

**Comment by vadimvitvickiy** [comment]
After trying to log in to the App Store Connect, I've got the same message, and had to unlock the account from my device.

**Comment by schwa** [comment]
And again today… 
<img width="412" alt="Screenshot 2024-07-09 at 08 10 48" src="https://github.com/XcodesOrg/xcodes/assets/20267/27c2ad4b-628a-4260-ac15-e25186d80103">

**Comment by halpz** [comment]
It locked my account again today - 
Can this work with app-specific passwords ?

**Comment by pmusolino** [comment]
Same issue here

**Comment by dfed** [comment]
I haven't seen this in a long time. I moved from using my personal Apple ID with Xcodes to using my work one.

**Comment by kct3937** [comment]
Suddenly XCodes has an error reading my Account info. Worked yesterday, but this morning...**hooped!**
Checked my AppleID and all is good.
Looks like XCodes did an update.

![CleanShot 2024-10-19 at 12 18 02](https://github.com/user-attachments/assets/b96b7d6a-ba11-4392-9ea4-8495bcc3f064)

**Comment by dfed** [comment]
This was likely fixed by #602 in 2.4.0

**Comment by nalexn** [comment]
At last 2.4.0 allowed me to log into apple account 🔥

**Comment by kct3937** [comment]
Mine is working too.  Good stuff!  Thanks to Xcodes

**Comment by dfed** [comment]
I'm guessing that this issue was fixed by https://github.com/XcodesOrg/xcodes/pull/393 and therefore can be closed.

**Comment by frankois944** [comment]
the issue is back!
Just locked 2 accounts in two days.
